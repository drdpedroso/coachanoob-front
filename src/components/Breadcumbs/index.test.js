import React from 'react';
import Breadcrumb from './index'
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'

const items = [
    {path: '/', text: 'Home'},
    {path: '/user', text: 'User'},
]
it('renders correctly', () => {
    const tree = renderer
        .create(<MemoryRouter initialEntries={['/']}><Breadcrumb items={items}></Breadcrumb></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should change breadcumb and set active', () => {
    const component = render(<MemoryRouter initialEntries={['/']}><Breadcrumb items={items}></Breadcrumb></MemoryRouter>)
    const lastBreadcrumb = component.getByTestId(items[1].text)
    expect(lastBreadcrumb.className.includes('active')).toBe(true)
});
