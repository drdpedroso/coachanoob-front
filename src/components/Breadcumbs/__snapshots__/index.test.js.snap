// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui breadcrumb"
>
  <a
    className="section"
    data-testid="Home"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <div
    className="divider"
  >
    /
  </div>
  <a
    className="active section"
    data-testid="User"
    href="/user"
    onClick={[Function]}
  >
    User
  </a>
</div>
`;
